### DISCUSSION TITLE
Documentation examples in TS vs JS

### DISCUSSION THREAD

**Discussion by ebonow** [JedWatson/react-select#4853] [General]
I wanted to get a general question out to the community: 
**Are the TypeScript examples and documentation a barrier to understanding how to use `react-select`?**

I have never worked in a professional environment that required type checking and while Flow and TypeScript are powerful, they likely aren't also a requirement or even familiarity for many. With Imposter Syndrome being so prevalent in our industry, I wanted to get some feedback to see if the mixture of JS vs TS examples was something that was silently affecting others. 

While I don't know what this would look like regarding examples or documentation, I know that the current documentation site is a huge obstacle to accessibility to many and wanted to know (at the risk of being exposed as someone who struggles at times to understand code written in Flow or TypeScript) if this is something that would help make this already complex library a little easier to grasp.

Perhaps an option to toggle a view of JS vs TS, I don't know, but simply wanted to start the discussion to know where others were with this potential concern.

**Comment by Methuselah96** [comment]
Great thought, having the ability to toggle between JS and TS would be really neat ([MUI](https://mui.com/) does this nicely).
I think it's important to keep the TS examples around because it provides good examples of how to use the types.

**Reply by ebonow** [reply]
Agreed.
